File: c-projects/CS50-introComputerScience/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    // Main labels
    float dollars;
    int change = 0;

    // Each coin type
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    // Prompt user for input using get_float
    do
    {
        dollars = get_float("Change owed: \n");
    }
    while (dollars <= 0);

    // Convert float input into int value
    int total = round(dollars * 100);

    // Loop for quarters
    while (total >= quarter)
    {
        total = total - quarter;
        change++;
    }

    // Loop for dimes
    while (total >= dime)
    {
        total = total - dime;
        change++;
    }
    
    // Loop for nickles
    while (total >= nickel)
    {
        total = total - nickel;
        change++;
    }

    // Loop for pennies
    while (total >= penny)
    {
        total = total - penny;
        change++;
    }

    // Print # of coins
    printf("%i\n", change);
}

File: python-projects/CS50-introComputerScience/pset6/cash.py
from cs50 import get_float


def main():
    change = 0
    # coin amounts
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1
    # prompt user input
    dollars = get_dollars()
    # convert float into int value
    total = round(dollars * 100)

    # loop for quarter
    while total >= quarter:
        total = total - quarter
        change += 1
    # loop for dime
    while total >= dime:
        total = total - dime
        change += 1
    # loop for nickel
    while total >= nickel:
        total = total - nickel
        change += 1
    # loop for penny
    while total >= penny:
        total = total - penny
        change += 1
    # print num of coins
    print(change)


def get_dollars():
    while True:
        d = get_float("Change owed: ")
        if d > 0:
            break
    return d


main()
